src/FluentValidation/Validators/StringEnumValidator.cs

```diff
@@ -0,0 +1,35 @@
+﻿namespace FluentValidation.Validators
+{
+	using System;
+	using System.Linq;
+	using System.Reflection;
+	using FluentValidation.Internal;
+	using Resources;
+
+	public class StringEnumValidator : PropertyValidator
+	{
+		private readonly Type _enumType;
+
+		private readonly bool _caseSensitive;
+
+		public StringEnumValidator(Type enumType) : this(enumType, false)
+		{
+		}
+
+		public StringEnumValidator(Type enumType, bool caseSensitive) : base(new LanguageStringSource(nameof(StringEnumValidator)))
+		{
+			_enumType = enumType;
+			_caseSensitive = caseSensitive;
+		}

```

Comment:
We need to check that the supplied type is an enum type, which would be best done in the constructor here. Perhaps throw an exception with the message "The type '{type.Name}' is not an enum and can't be used with IsEnumName"